torch/_dynamo/symbolic_convert.py

```diff
@@ -729,19 +729,56 @@ def step(self):
 
             log.debug("step triggered compile", exc_info=True)
 
-        # generate code from checkpoint
+        # generate code from checkpoint 
         assert not self.output.output_instructions
         assert self.checkpoint is not None
         continue_inst, state = self.checkpoint
+
+        cannot_restore_entry = next(
+            (entry for entry in reversed(self.block_stack) if not entry.can_restore()),
+            None,
+        )
+        last_jump_forward = None
+        if cannot_restore_entry and not self.one_graph:
+            cannot_restore_target = cannot_restore_entry.target
+            insts = self.instructions[self.instruction_pointer :]
+            in_active_region = False
+            for inst in insts:
+                if inst == cannot_restore_target:
+                    in_active_region = True
+                if in_active_region and inst.opname == "JUMP_FORWARD":
+                    last_jump_forward = inst
+
         self.restore_graphstate(state)
         self.output.compile_subgraph(
             self,
+            # TODO: not sure if this needs to be changed if we are going to resume_at
             partial_convert=True,
             reason=GraphCompileReason("step_unsupported", [self.frame_summary()]),
         )
-        self.output.add_output_instructions(
-            [create_jump_absolute(continue_inst)] + self.instructions
-        )
+        # The reason why we fail is due to non-restorable block stack entries.
+        # If a jump target in its unwind/cleanup block exists, resume at the jump target.
+        if last_jump_forward:
+            target_index = None
+            for idx, instruction in enumerate(self.instructions):
+                if instruction == last_jump_forward.target:
+                    target_index = idx
+                    break
+            assert target_index is not None
+
+            resume_at = self.create_call_resume_at(last_jump_forward.target)
```

Comment:
Unfortunately, the stack/locals is wrong for this call.

Reply:
It is impossible to know what the stack and locals would be at the time of making this call.

Reply:
Furthermore - the stack and locals at this time is non-deterministic.

That's because the eagerly executed code is non-deterministic. This means that we need to _guard on the stack and locals_ at the time of the call.

Reply:
Still, while the general case seems hard, it does seem that this code ought to be traceable in some special cases, and furthermore, it seems likely that users ought to have try/excepts with assertions thrown in their code, so being able to handle this does seem quite valuable.

Reply:
The typical pattern is probably:

```python
try:
   # some torch code
except:
  # some other torch code
```

So we could probably convert this into a jump, as long as `check_exc_match` can match any exception... :thinking: